is_core/auth_token/utils.py

```diff
@@ -44,13 +60,21 @@ def logout(request):
         user = None
     user_logged_out.send(sender=user.__class__, request=request, user=user)
 
-    if hasattr(request, 'user'):
-        request.user = AnonymousUser()
-
     if hasattr(request, 'token') and request.token.is_active:
-        token = request.token
-        token.is_active = False
-        token.save()
+        if request.token.active_takeover:
```

Comment:
You should save it to variable and than check it, now there would be two calls to database for same thing.